### PR TITLE
add zarr recipe

### PR DESCRIPTION
**Pull request by alimanfoo**

### CODE REVIEW

**Comment by alimanfoo** [comment]
It looks like the fasteners package which is a runtime dependency of zarr is not available in the default conda channels. It is available on anaconda from [chilipp/fasteners](http://anaconda.org/chilipp/fasteners). Can you advise on the best way to satisfy this dependency?

File: recipes/zarr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  git_url: https://github.com/alimanfoo/zarr.git
+  git_tag: v{{ version }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - python
+    - pip
```
Comment (ocefpaf) [inline recipes/zarr/meta.yaml]: You probably don't need `pip`.

File: recipes/zarr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+package:
+  name: zarr
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/alimanfoo/zarr.git
+  git_tag: v{{ version }}
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/zarr/meta.yaml]: You can a line here with `script: python setup.py install --single-version-externally-managed --record record.txt` and remove both `bld.bat` and `build.sh` files.

**Comment by ocefpaf** [comment]
> It looks like the fasteners package which is a runtime dependency of zarr is not available in the default conda channels. It is available on anaconda from [chilipp/fasteners](http://anaconda.org/chilipp/fasteners). Can you advise on the best way to satisfy this dependency?

You can download the [tarball](http://anaconda.org/Chilipp/fasteners/0.14.1/download/linux-64/fasteners-0.14.1-py27_0.tar.bz2) from that channel and adapt the recipe you will find in the `info/recipe` directory. It is also nice to contact the author of the original recipe and invite him to be a maintainer of the recipe here at conda-forge.

**Comment by alimanfoo** [comment]
Thanks. Should I include the fasteners recipe within this PR or make a
separate PR?

On Tuesday, 17 May 2016, Filipe <notifications@github.com> wrote:

> It looks like the fasteners package which is a runtime dependency of zarr
> is not available in the default conda channels. It is available on anaconda
> from chilipp/fasteners <http://anaconda.org/chilipp/fasteners>. Can you
> advise on the best way to satisfy this dependency?
>
> You can download the tarball
> <http://anaconda.org/Chilipp/fasteners/0.14.1/download/linux-64/fasteners-0.14.1-py27_0.tar.bz2>
> from that channel and adapt the recipe you will find in the info/recipe
> directory. It is also nice to contact the author of the original recipe and
> invite him to be a maintainer of the recipe here at conda-forge.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-219708163>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by ocefpaf** [comment]
> Thanks. Should I include the fasteners recipe within this PR or make a
separate PR?

It is fine to add it here. We don't recommend adding tons of recipes in one go, but 2-3 is fine.

**Comment by alimanfoo** [comment]
@ocefpaf I've added recipes for fasteners (dependency of zarr) and monotonic (dependency of fasteners). Looks like builds are passing now.

File: recipes/fasteners/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - monotonic >=0.1
+
+test:
+  imports:
+    - fasteners
+    - fasteners.tests
+
+about:
+  home: https://github.com/harlowja/fasteners
+  license: Apache Software License
```
Comment (ocefpaf) [inline recipes/fasteners/meta.yaml]: Do you know which version of the Apache license? `Apache-2.0`?

File: recipes/monotonic/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  run:
+    - python
+
+test:
+  imports:
+    - monotonic
+
+about:
+  home: https://github.com/atdt/monotonic
+  license: Apache Software License
```
Comment (ocefpaf) [inline recipes/monotonic/meta.yaml]: Same question as above.

**Comment by ocefpaf** [comment]
@alimanfoo everything looks good. Just a minor question regarding the license version and this is good to go.

**Comment by alimanfoo** [comment]
Thanks @ocefpaf, licenses fixed.

**Comment by alimanfoo** [comment]
Thanks @ocefpaf. Can I ask, what is the recommended process for inviting the developers of monotonic and fasteners to become maintainers of the respective feedstocks?

**Comment by ocefpaf** [comment]
There is no hard rule. Basically you have to explain to them what is conda-forge and what is expect of the maintainer, like merge an eventual maintenance PR, or updating the to the latest release .

**Comment by jakirkham** [comment]
If you would like, you can steal one of my old comments and tweak them. Here is [one]( https://github.com/conda-forge/staged-recipes/pull/256#issuecomment-205851984 ) I did for `caffe`. Would add that it helps to know your audience. For instance, it is not worth explaining `conda` to someone that uses it regularly.

**Comment by alimanfoo** [comment]
Thanks both. What is the process for adding someone to a conda-forge team,
e.g., the fasteners team?

On Wednesday, May 18, 2016, jakirkham <notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:

> If you would like, you can steal one of my old comments and tweak them.
> Here is one
> <https://github.com/conda-forge/staged-recipes/pull/256#issuecomment-205851984>
> I did for caffe. Would add that it helps to know your audience. For
> instance, it is not worth explaining conda to someone that uses it
> regularly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-220046149>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by ocefpaf** [comment]
@alimanfoo you need to add their GitHub handle as maintainers via a PR to the recipe/team you want them to be a part of.

**Comment by alimanfoo** [comment]
Sorry, not quite following. Do you mean I need to add their github handle
into the "recipe-maintainers" tag in the meta.yaml file within the
feedstock repo?

On Wednesday, May 18, 2016, Filipe <notifications@github.com> wrote:

> @alimanfoo <https://github.com/alimanfoo> you need to add their GitHub
> handle as maintainers via a PR to the recipe/team you want them to be a
> part of.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-220113216>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by jakirkham** [comment]
Yes, exactly. Make sure to make this change in your fork and PR it back to the main repo.

**Comment by alimanfoo** [comment]
OK, thanks. For other changes to the recipe like increasing the version
number, it's OK to push directly to the main repo, or should I always
submit a PR from my own fork?

On Wednesday, May 18, 2016, jakirkham <notifications@github.com> wrote:

> Yes, exactly. Make sure to make this change in your fork and PR it back to
> the main repo.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-220129129>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by ocefpaf** [comment]
@alimanfoo you should always submit PRs, but you don't need to bump the builder number nor wait for CIs when adding/removing maintainers.

**Comment by jakirkham** [comment]
Also it is good practice to get them to sign off somehow.

**Comment by alimanfoo** [comment]
Sorry, what do you mean by sign off?

On Wednesday, 18 May 2016, jakirkham <notifications@github.com> wrote:

> Also it is good practice to get them to sign off somehow.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-220140106>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by jakirkham** [comment]
Alright, let me try to rephrase. Please only add people as maintainers when they have given their consent. Does that make sense?

**Comment by alimanfoo** [comment]
Ah, yes of course.

On Thursday, 19 May 2016, jakirkham <notifications@github.com> wrote:

> Alright, let me try to rephrase. Please only add people as maintainers
> when they have given their consent. Does that make sense?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/627#issuecomment-220204465>
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Pull request merged**